libs/cart/guides/extension.md

```diff
@@ -0,0 +1,106 @@
+# Extension
+
+`@daffodil/cart` provides a number of extension mechanisms so that it can be customized to fit specific needs.
+
+## Generic Models
+
+All Daffodil layers can operate on generic extensions of vanilla Daffodil models. Custom models can therefore be used while retaining type safety. The following example illustrates customizing the cart model with the cart facade.
+
+```typescript
+import {
+  DaffCartFacade,
+  DaffCart
+} from '@daffodil/cart';
+
+interface CustomCart extends DaffCart {
+  customField: string;
+}
+
+@Component({})
+class CartComponent implements OnInit {
+  cart$: Observable<CustomCart>;
+
+  constructor(private cartFacade: DaffCartFacade<CustomCart>) {}
+
+  ngOnInit() {
+    // this.cartFacade.cart$ is of type CustomCart
+    this.cart$ = this.cartFacade.cart$;
+  }
+}
+```
+
+## Extensible GraphQL Fragments
+
+Arbitrary additional fields can be requested on the cart object. Inject a GraphQL document node containing fragments on the platform's cart type to define extra fields.
+
+The following cart drivers support extensible fragments:
+- Magento
+
+### Magento
+
+Provide the `DaffMagentoExtraCartFragments` to query additional fields on a Magento cart query. This applies to all of the driver calls that return a `DaffCart`, which is most of them.
```

Comment:
This can work with all queries, right?

Reply:
Not queries that don't return the cart, like `placeOrder`

Reply:
why can't we do this:

```typescript
import gql from 'graphql-tag';
import {
  DaffMagentoExtraOrderFragments,
  DaffCartFacade,
  DaffCartLoad,
  DaffCart
} from '@daffodil/cart';
const extraOrderFragment = gql`
  fragment ExtraCartFields on Order {
    field1 {
      amount
    }
    shipping_addresses {
      field2 {
        value
      }
    }
  }
`;
@NgModule({
  ...,
  providers: [
    {
      provide: DaffMagentoExtraOrderFragments,
      useValue: extraOrderFragment,
      multi: true
    }
  ]
})
class AppModule {}
```

or something like that

Reply:
I mean, we could and should for the order package, but that doesn't belong here. `placeOrder` returns a `DaffCartOrderResult` that is neither a cart nor an order.

Reply:
I was just using the example you provided. My point was just that I think we can do this with all models and all queries, which is at odds with "This applies to all of the driver calls that return a `DaffCart`, which is most of them."

Reply:
We can do this with any fragment but currently the only one that we have implemented is the Cart one. I'm not trying to document general extension mechanisms but only the ones that currently exist in the cart package.

Reply:
ok sounds good